java-checks/src/main/java/org/sonar/java/checks/MathClampMethodsCheck.java

```diff
@@ -0,0 +1,183 @@
+
+        return (isLessThanOrEqual(innerExpression.condition())
+          && checkInnerExpression(innerCondition, innerExpression.trueExpression(), innerExpression.falseExpression(), isMax))
+          || (isGreaterThanOrEqual(innerExpression.condition())
+            && checkInnerExpression(innerCondition, innerExpression.trueExpression(), innerExpression.falseExpression(), !isMax));
+      } else {
+        return tree2.is(Tree.Kind.METHOD_INVOCATION) && (matcher.matches((MethodInvocationTree) tree2));
+      }
+    }
+    return false;
+  }
+
+  private static boolean checkInnerExpression(BinaryExpressionTree innerCondition, ExpressionTree innerTrueExpression, ExpressionTree innerFalseExpression, boolean isMax) {
+    if (innerCondition.leftOperand().is(Tree.Kind.IDENTIFIER) && innerCondition.rightOperand().is(Tree.Kind.IDENTIFIER)
+      && innerTrueExpression.is(Tree.Kind.IDENTIFIER) && innerFalseExpression.is(Tree.Kind.IDENTIFIER)) {
+      var leftOperandName = ((IdentifierTree) innerCondition.leftOperand()).name();
+      var rightOperandName = ((IdentifierTree) innerCondition.rightOperand()).name();
+      var trueExpressionName = ((IdentifierTree) innerTrueExpression).name();
+      var falseExpressionName = ((IdentifierTree) innerFalseExpression).name();
+
+      return isMax ? (leftOperandName.equals(falseExpressionName) && rightOperandName.equals(trueExpressionName))
+        : (leftOperandName.equals(trueExpressionName) && rightOperandName.equals(falseExpressionName));
+    }
+    return false;
+  }
+
+  private void checkMethodInvocation(MethodInvocationTree tree) {
+    if (MATH_MIN_METHOD_MATCHERS.matches(tree)) {
+      var firstArg = tree.arguments().get(0);
+      var secondArg = tree.arguments().get(1);
+
+      if ((firstArg.is(Tree.Kind.METHOD_INVOCATION) && MATH_MAX_METHOD_MATCHERS.matches((MethodInvocationTree) firstArg)) ||
+        (secondArg.is(Tree.Kind.METHOD_INVOCATION) && MATH_MAX_METHOD_MATCHERS.matches((MethodInvocationTree) secondArg))) {
+        reportIssue(tree, METHOD_INVOCATION_MESSAGE);
+      }
+    }
+    if (MATH_MAX_METHOD_MATCHERS.matches(tree)) {
+      var firstArg = tree.arguments().get(0);
+      var secondArg = tree.arguments().get(1);
+
+      if ((firstArg.is(Tree.Kind.METHOD_INVOCATION) && MATH_MIN_METHOD_MATCHERS.matches((MethodInvocationTree) firstArg)) ||
+        (secondArg.is(Tree.Kind.METHOD_INVOCATION) && MATH_MIN_METHOD_MATCHERS.matches((MethodInvocationTree) secondArg))) {
+        reportIssue(tree, METHOD_INVOCATION_MESSAGE);
+      }
+    }
+  }
+
+  private void checkIfStatement(IfStatementTree tree) {
+    if (isGreaterThanOrLessThan(tree.condition())
+      && (tree.elseStatement() != null && tree.elseStatement().is(Tree.Kind.IF_STATEMENT))) {
+      var elseIfStatement = (IfStatementTree) tree.elseStatement();
+      if (isGreaterThanOrLessThan(elseIfStatement.condition())) {
+        reportIssue(tree, IF_ELSE_STATEMENT_MESSAGE);
+      }
+    }
+  }
```

Comment:
The logic about `if` statement is too basic. It will catch a lot of false-positives. For example:
```
    int a,b,c,d,e,f;
    if (a < b) { // Noncompliant, false positive
      System.out.println("no assignment");
    } else if (c < d) { // Noncompliant, false positive
      System.out.println("no assignment");
    } else if (e < f) {
      System.out.println("no assignment");
    }
```

Reply:
Let's remove the `if` from the scope of this rule.